backend/src/main.rs

```diff
@@ -91,11 +91,21 @@ struct UserState {
 }
 
 impl UserState {
-    pub fn send(&self, msg: &GameMessage) {
-        if let Ok(s) = serde_json::to_string(msg) {
-            let _ = self.tx.send(Ok(Message::text(s)));
+    pub async fn send(&self, msg: &GameMessage) -> bool {
+        send_to_user(&self.tx, msg).await
+    }
+}
+
+async fn send_to_user(
+    tx: &'_ mpsc::UnboundedSender<Result<Message, warp::Error>>,
+    msg: &GameMessage,
+) -> bool {
+    if let Ok(s) = serde_json::to_string(msg) {
+        if let Ok(compressed) = zstd::stream::encode_all(&mut std::io::Cursor::new(s), 0) {
```

Comment:
you're not actually using the streaming API so might be simpler to use `zstd::block`?

Reply:
will check that out!

I've been hesitant to land this because I've been trying to see if I can do permessage-deflate instead at the proxy layer (i.e. in NGINX), and haven't swung back around to actually see if it works